### DISCUSSION TITLE
Filter gauge transducer for Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by spifftek70** [tagyoureit/nodejs-poolController#995] [Nixie]
I have a Raspberry Pi 4, stacked with Sequent Mega-Bas and 8-Relay board. I want to add transducers for both filters (pool and spa).

Will this "[RPi 3.3V to 5V 26 I/O Bidirectional Voltage-Level Shifter Module for Raspberry Pi, Logic Level Converter](https://www.amazon.com/Bidirectional-Voltage-Level-Shifter-Raspberry-Converter/dp/B0979BSYCW/ref=asc_df_B0979BSYCW/?tag=hyprod-20&linkCode=df0&hvadid=692875362841&hvpos=&hvnetw=g&hvrand=4398813932322713388&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9199160&hvtargid=pla-2281435177378&psc=1&mcid=c0702d6a3e8333ada66fc7f6d0965e07&hvocijid=4398813932322713388-B0979BSYCW-&hvexpln=73&gad_source=1)" work? From what I've read, It will process the analog to digital, plus the level shifting for 3.3v. 

Or perhaps this "[4-Channel 16-Bit ADC for Raspberry Pi (ADS1115)](https://wiki.seeedstudio.com/4-Channel_16-Bit_ADC_for_Raspberry_Pi-ADS1115/
)"?
Thoughts?

[Transducer](https://www.amazon.com/Pressure-Transducer-Sender-Sensor-Connector/dp/B08FR7G4QB/ref=sr_1_6?crid=1YTWZEO4KP3CX&dib=eyJ2IjoiMSJ9.oisg8znA1OzvYGyxMBGedibhXnQG_c8B8r-hIY2UN2rjptS_mIUzXvEIziXIK5HZ977VtGROvoLX9NzpWEn8xVaohXn4pc14kPbeoMXBnb9cdviyqKmh5CtsFKMxSpsU-GpaubZyRbYGJvG5hXYCnjXIgMrHozgbWBOWbeVSjnBjF_l4o5zRmk-RTvqYupuCLgIGcBKS81HU92CpYjolplw0p-zfnJsBasc2sb0c3A4.2AKIibax8JvhVwcIcKWOs3CJ0ezHSqxSXQ8JkuYAnsk&dib_tag=se&keywords=100+psi+water+transducer+1%2F4%22&qid=1720544970&sprefix=100+psi+water+transducer+1%2F4+%2Caps%2C132&sr=8-6) in mind

**Comment by rstrouse** [comment]
It should by why not the 0-10v input on the BAS.  I don't know how to wire it but it should work.

**Reply by spifftek70** [reply]
It does???  I can't find anyone who's wired it....

**Reply by rstrouse** [reply]
Ask the folks at Sequent how to hook up a 5v transducer.

**Reply by tagyoureit** [reply]
I had one identical to this.  Just wire the 5v output to the 0-10v input on the MegaBas.

The input here is on input 4.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/aedb34fb-664a-4e4f-8dca-6460dca2158e)

Feed the voltage to a generic pressure transducer device.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/c4bafbdd-a0de-44db-96d4-6fe3d800dc5c)

And setup the PSI/volt.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/ac260c23-a1d0-44d6-93fe-cc7fb55417a2)

And finally feed the pressure value to njsPC.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/032a03e2-6590-4d48-9bc0-98aeb3841269)

**Reply by spifftek70** [reply]
There is a Red (5v) cable, a black (ground) cable, and a green (data?) cable.  black and red are go to the 10v and ground ports in the Mega-Bas.  What do I do with the green cable?

**Reply by tagyoureit** [reply]
Green will be your 0.5-4.5v output.

**Comment by spifftek70** [comment]
Something like this?

![smb](https://github.com/tagyoureit/nodejs-poolController/assets/11727190/5bce77bf-a15b-4e66-8047-3a5bf36d0d12)

**Comment by spifftek70** [comment]
It's working.  thanks!

**Comment by EPiSKiNG** [comment]
Apologies for resurrecting an old thread, but I'm working on setting up a pressure transducer on my Sequent MEGA-BAS.  Is the wiring diagram listed above valid?  The ground needs to be shared between the 0-10V Output and the Input side?  I can't seem to get anything other than 0.005V reading when the pump is running or off.  

Thanks!

**Comment by EPiSKiNG** [comment]
Ahh, I figured it out!  The wiring diagram is correct, I just needed to configure the I2C output to 5V, where previously it was disabled :)

<img width="795" height="299" alt="image" src="https://github.com/user-attachments/assets/41c1f04c-fbda-426b-b774-69dd5774dcbd" />
